internal/security/bootstrapper/messagebus/handlers/handlers.go

```diff
@@ -0,0 +1,181 @@
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed under the License
+* is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+* or implied. See the License for the specific language governing permissions and limitations under
+* the License.
+*
+*******************************************************************************/
+
+package handlers
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"html/template"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/messagebus/container"
+
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/secret"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	bootstrapConfig "github.com/edgexfoundry/go-mod-bootstrap/v2/config"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+)
+
+const (
+	// file read/write permission only for owner
+	readWriteOnlyForOwner os.FileMode = 0600
+)
+
+// Handler is the redis bootstrapping handler
+type Handler struct {
+	credentials bootstrapConfig.Credentials
+}
+
+// NewHandler instantiates a new Handler
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+// GetCredentials retrieves the message bus credentials from secretstore
+func (handler *Handler) GetCredentials(ctx context.Context, _ *sync.WaitGroup, startupTimer startup.Timer,
+	dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	config := container.ConfigurationFrom(dic.Get)
+	secretProvider := bootstrapContainer.SecretProviderFrom(dic.Get)
+
+	var credentials = bootstrapConfig.Credentials{
+		Username: "unset",
+		Password: "unset",
+	}
+
+	for startupTimer.HasNotElapsed() {
+		// retrieve message bus credentials from secretstore
+		secrets, err := secretProvider.GetSecret(config.MessageQueue.SecretName)
+		if err == nil {
+			credentials.Username = secrets[secret.UsernameKey]
+			credentials.Password = secrets[secret.PasswordKey]
+			break
+		}
+
+		lc.Warnf("Could not retrieve message bus credentials (startup timer has not expired): %s", err.Error())
+		startupTimer.SleepForInterval()
+	}
+
+	if credentials.Password == "unset" {
+		lc.Error("Failed to retrieve message bus credentials before startup timer expired")
+		return false
+	}
+
+	handler.credentials = credentials
+	return true
+}
+
+// SetupPasswordFile sets up the message bus password file for MQTT specific
+func (handler *Handler) SetupPasswordFile(ctx context.Context, _ *sync.WaitGroup, startupTimer startup.Timer,
+	dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	config := container.ConfigurationFrom(dic.Get)
+
+	if config.MessageQueue.Type != "mqtt" {
+		lc.Infof("messaage bus not mqtt type: %s, skip setting up password file", config.MessageQueue.Type)
+		return true
+	}
```

Comment:
Why? this will only run from the entry point script on `mosquitto` so already know it is MQTT